### ISSUE TITLE
no-magic-number regression starting with v20.1.0

### ISSUE THREAD

**Issue by scottohara** [eslint/typescript-eslint-parser#588]
**What version of ESLint are you using?**
5.11.1

**What version of TypeScript are you using?**
3.1.1

**What version of `typescript-eslint-parser` are you using?**
Any version from 20.1.0 onwards

**What code were you trying to parse?**

```json
{
  "parser": "typescript-eslint-parser",
  "rules": {
    "no-magic-numbers": "error"
  }
}
```

```ts
type Amount = 10 | -10;

const amount: Amount = 10;
```

**What did you expect to happen?**
No errors

**What happened?**
In `typescript-eslint-parser@20.0.0`, code passes with no errors.

With `typescript-eslint-parser@20.1.0` onwards, the errors below occur:

```bash
1:15  error  No magic number: 10   no-magic-numbers
1:20  error  No magic number: -10  no-magic-numbers
```

The magic numbers that the rule is complaining about are the allowed values of the Typescript numeric literal type `type Amount = 10 | -10`.

I'm not aware of any way to define a numeric literal type that doesn't violate the magic numbers rule?

**Comment by armano2** [comment]
issue is solved in #584

**Comment by JamesHenry** [comment]
584 turned into #596 and has been merged
